File: src/main/java/corp/soyummy/recipefavorites/dto/RecipeFavoriteCreateDto.java
package corp.soyummy.recipefavorites.dto;

import lombok.Getter;
import lombok.Setter;
import lombok.AllArgsConstructor;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import jakarta.validation.constraints.Size;

@Getter
@Setter
@AllArgsConstructor
public class RecipeFavoriteCreateDto {
    @NotNull(message = "Recipe ID is required!")
    @Size(min = 1, message = "RecipeId must have at least one character.")
    private String recipeId;

    @NotNull(message = "Amount is required!")
    @Positive(message = "Amount must be a positive number!")
    private Integer amount;
}

File: src/main/kotlin/com/soyummy/recipefavorites/dto/RecipeFavoriteCreateDto.kt
package com.soyummy.recipefavorites.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Positive

data class RecipeFavoriteCreateDto(
    @field:NotBlank(message = "Recipe ID is required!")
    val recipeId: String,

    @field:NotNull(message = "Amount is required!")
    @field:Positive(message = "Amount must be a positive number!")
    val amount: Int
)